### PR TITLE
Add to_xml for serializing a model to XML

### PR DESCRIPTION
**Pull request by jmuhlich**
Now with backport of `ElementTree.canonicalize` from 3.8!

### CODE REVIEW

**Comment by jmuhlich** [comment]
`rebase -f` followed by `push -f` finally got the commits displaying here in the right order. Hopefully it was just a cosmetic issue on the website but it's still concerning.

**Comment by jmuhlich** [comment]
Oh, of course I need to add `to_xml` to the Usage section of the docs.

**Review by tlambert03 — APPROVED** [review]
@jmuhlich, This is really amazing... immediately makes this library _much_ more useful.  great job!  Feel free to merge whenever you're ready.  We can add usage notes to the docs in a follow up PR (or here, if you'd prefer).

@JacksonMaxfield, I believe you were interested in this functionality.  This has all come a long way since you last checked in... so I'd be curious to see if this fits your application at this point.

**Comment by JacksonMaxfield** [comment]
@tlambert03 thanks for the alert! Will toss this back to the team. We are actively deciding how to handle OME in aicsimageio 4.0.0 so this perfect timing.

File: docs/source/usage.rst [hunk 1]
```diff
@@ -83,25 +84,87 @@ For example, you can parse an ome.xml, and then explore it with pythonic,
     In [8]: ome.images[0].pixels.channels[0].emission_wavelength                                                                               
     Out[8]: 523.0
 
+Then you might want to make some changes:
```
Comment (tlambert03) [inline docs/source/usage.rst:87]: just love all of this 👇

**Comment by jmuhlich** [comment]
I modified the Usage docs to demonstrate `to_xml`. If you like this approach we can merge this.

I realized the round-trip test could miss certain regressions due to the normalization/canonicalization step. And the normalization is problematic for XMLAnnotations anyway, so I was thinking of crafting a new XML example that is expected to round-trip faithfully without that step. I'll do that in a new PR.

**Comment by tlambert03** [comment]
> I modified the Usage docs to demonstrate `to_xml`. If you like this approach we can merge this.

yeah, I love it. I have thoughts for how to expand the examples, but lets merge and move on.  This opens up lots of new avenues. 

Not sure if you saw my PR to your branch: https://github.com/jmuhlich/ome-types/pull/1 ... but I think I'd ultimately like `from_xml` and `to_xml` to be methods on `OME`.  thoughts on that?

> I realized the round-trip test could miss certain regressions due to the normalization/canonicalization step. And the normalization is problematic for XMLAnnotations anyway, so I was thinking of crafting a new XML example that is expected to round-trip faithfully without that step. I'll do that in a new PR.

sounds good to me.  we've got another bug (#45) that also needs fixing, but would be nice to get this in first

**Comment by jmuhlich** [comment]
I saw the change to add the methods, will discuss there.

**Pull request merged**